src/components/Characters.js

```diff
@@ -1,7 +1,51 @@
+import "./Characters.css";
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+
 export default function Characters() {
+  const [characters, setCharacters] = useState([]);
+  const [page, setPage] = useState(1);
+  const [totalPages, setTotalPages] = useState();
+
+  useEffect(() => {
+    const url = `https://rickandmortyapi.com/api/character/?page=${page}`;
+
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        setCharacters((prevCharacters) => {
+          return [...prevCharacters, ...data.results];
+        });
+        console.log(data);
+        setTotalPages(data.info.pages);
+      });
+  }, [page]);
+
+  function renderCharacters() {
+    return characters.map((character) => {
+      const { id, name, image } = character;
+
+      return (
+        <li key={id} className="characterBox">
+          <Link to={`/characters/${id}`}>
+            <img className="avatar" src={image} alt={name} />
+            <p className="characterName">{name} </p>
+          </Link>
+        </li>
+      );
```

Comment:
A refactor idea: This looks like a Component to me, that you can return
```suggestion
      return <CharacterBox key={id} url={`/characters/${id}`} src={img} name={name} />
```
You need to create the component and import it in this file. 
What do you think? Makes this sense?